x/mongo/driver/connstring/connstring.go

```diff
@@ -340,6 +340,9 @@ func (p *parser) validate() error {
 		if p.Scheme == SchemeMongoDBSRV {
 			return errors.New("a direct connection cannot be made if an SRV URI is used")
 		}
+		if p.LoadBalancedSet && p.LoadBalanced {
```

Comment:
Makes more sense to have this check here since its an "invalid use of direct connections".